controller/server/main.go

```diff
@@ -183,35 +192,73 @@ func validatePath(path string) (string, error) {
 
 func (s *server) CreateTopology(ctx context.Context, req *cpb.CreateTopologyRequest) (*cpb.CreateTopologyResponse, error) {
 	log.Infof("Received CreateTopology request: %v", req)
-	bp, err := validatePath(req.Topology.Name)
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "invalid topology path: %v", err)
+	topoPb := req.GetTopology()
+	if topoPb == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid request: missing topology protobuf")
 	}
-	p := topo.TopologyParams{
-		TopoName:       req.Topology.Name,
-		TopoNewOptions: []topo.Option{topo.WithBasePath(filepath.Dir(bp))},
-		Kubecfg:        req.Kubecfg,
+	if topoPb.GetName() == "" {
+		return nil, status.Errorf(codes.NotFound, "missing topology name")
+	}
+
+	s.muTopo.Lock()
+	defer s.muTopo.Unlock()
+	if _, ok := s.topos[topoPb.GetName()]; ok {
+		return nil, status.Errorf(codes.AlreadyExists, "topology %q already exists", req.Topology.GetName())
+	}
+
+	log.Infof("Validating the topology protobuf...")
+	for _, node := range topoPb.Nodes {
+		if node.GetType() == tpb.Node_IXIA_TG {
+			continue
+		}
+		if node.GetConfig() == nil {
+			return nil, status.Errorf(codes.NotFound, "missing config field for node %q", node.GetName())
```

Comment:
above we are returning codes.InvalidArgument for when a specified file is not found.

Which code do you think is better for this?

I would lean towards invalid argument here

Reply:
After checking https://grpc.github.io/grpc/core/md_doc_statuscodes.html, I think invalid argument makes more sense. Done.